packages/atomic/cypress/integration/smart-snippet-feedback-modal.cypress.ts

```diff
@@ -0,0 +1,173 @@
+import {addSmartSnippet} from './smart-snippet-actions';
+import {SmartSnippetSelectors} from './smart-snippet-selectors';
+import * as SmartSnippetFeedbackModalAssertions from './smart-snippet-feedback-modal-assertions';
+import * as CommonAssertions from './common-assertions';
+import {
+  smartSnippetFeedbackModalComponent,
+  SmartSnippetFeedbackModalSelectors,
+} from './smart-snippet-feedback-modal-selectors';
+
+describe('Smart Snippet Feedback Modal Test Suites', () => {
+  function setupOpenModal() {
+    new TestFixture().with(addSmartSnippet()).init();
+    SmartSnippetSelectors.feedbackDislikeButton().click();
+    SmartSnippetSelectors.feedbackExplainWhy().click();
+  }
+
+  describe('after opening the modal', () => {
+    before(() => {
+      setupOpenModal();
+    });
+
+    SmartSnippetFeedbackModalAssertions.assertDisplayModal(true);
+    SmartSnippetFeedbackModalAssertions.assertDisplayDetails(false);
+    CommonAssertions.assertAccessibility(smartSnippetFeedbackModalComponent);
+
+    it('should give the primary text color to the cancel button', () => {
+      SmartSnippetFeedbackModalSelectors.cancelButton()
+        .invoke('css', 'color')
+        .should('eq', 'rgb(48, 63, 159)');
+    });
+
+    describe('then clicking submit', () => {
+      before(() => {
+        SmartSnippetFeedbackModalSelectors.submitButton().click();
+      });
+
+      SmartSnippetFeedbackModalAssertions.assertFormErrors();
+      SmartSnippetFeedbackModalAssertions.assertDisplayModal(true);
+    });
+  });
+
+  describe('after opening the modal then selecting some option except other', () => {
+    before(() => {
+      setupOpenModal();
+      SmartSnippetFeedbackModalSelectors.reasonRadio().first().click();
+    });
+
+    SmartSnippetFeedbackModalAssertions.assertDisplayDetails(false);
+
+    describe('then clicking submit', () => {
+      before(() => {
+        SmartSnippetFeedbackModalSelectors.submitButton().click();
+      });
+
+      SmartSnippetFeedbackModalAssertions.assertDisplayModal(false);
+    });
+  });
+
+  describe('after opening the modal then selecting other', () => {
+    before(() => {
+      setupOpenModal();
+      SmartSnippetFeedbackModalSelectors.reasonRadio().last().click();
+    });
+
+    SmartSnippetFeedbackModalAssertions.assertDisplayDetails(true);
+
+    describe('then clicking submit', () => {
+      before(() => {
+        SmartSnippetFeedbackModalSelectors.submitButton().click();
+      });
+
+      SmartSnippetFeedbackModalAssertions.assertFormErrors();
+      SmartSnippetFeedbackModalAssertions.assertDisplayModal(true);
+
+      describe('then writing details and pressing submit', () => {
+        before(() => {
+          SmartSnippetFeedbackModalSelectors.detailsInput().type('abc', {
+            force: true,
+          });
+          SmartSnippetFeedbackModalSelectors.submitButton().click();
+        });
+
+        SmartSnippetFeedbackModalAssertions.assertDisplayModal(false);
```

Comment:
I don't know if you plan on making this part of a separate PR, but I think that after you submit your reason, the 'Explain why' button should disappear

Reply:
I didn't plan to add that, I vaguely recall a discussion on whether or not we should do that, but I don't remember the conclusion 🤔  I'll check

Reply:
so you could send the explanation multiple times?

Reply:
update: discussed it again, and you're correct, the button should disappear. I'll update the PR.